core/domain/user_services.py

```diff
@@ -310,12 +310,18 @@ def update_user_bio(user_id, user_bio):
     _save_user_settings(user_settings)
 
 
-def update_first_contribution_datetime(user_id, first_contribution_datetime):
+def _update_first_contribution_datetime_if_necessary(user_id, first_contribution_datetime):
```

Comment:
This method should be public and it should call update_first_contribution_datetime, which should be private.